bochs/cpu/cpu.h

```diff
@@ -5500,7 +5503,7 @@ class bxInstruction_c;
 
 #define BX_NEXT_INSTR(i) {                             \
   BX_COMMIT_INSTRUCTION(i);                            \
-  if (BX_CPU_THIS_PTR async_event) return;             \
+  if (BX_CPU_THIS_PTR async_event || bx_dbg.debugger_active) return;             \
```

Comment:
we basically not allow traces when bx_dbg.debugger_active is set because debugger doesn't work well with them (many things are checked in main loop and not in the instruction handlers).
probably the easiest way to stop trace everywhere will be use smth like STOP_TRACE bit in async_event
May be invent one called ASYNC_EVENT_IN_DEBUGGER, set it from cpu_loop_debugger and will automatically cause exist to main loop after every instruction due to async_event being set

Mainly I want all this to avoid the "|| bx_dbg.debugger_active" here which actually reduce emulation speed a bit (~1% measured) but also code will be much simpler

Reply:
done